dissect/util/compression/xz.py

```diff
@@ -0,0 +1,101 @@
+import io
+from binascii import crc32
+from typing import BinaryIO
+
+from dissect.util.stream import OverlayStream
+
+
+def repair_checksum(fh: BinaryIO) -> BinaryIO:
+    """Repair CRC32 checksums for all headers in an XZ stream.
+
+    FortiOS XZ files have (on purpose) corrupt streams which they read using a modified ``xz`` binary.
+    The only thing changed are the CRC32 checksums, so partially parse the XZ file and fix all of them.
+
+    References:
+        - https://tukaani.org/xz/xz-file-format-1.1.0.txt
+        - https://github.com/Rogdham/python-xz
+
+    Args:
+        fh: A file-like object of an LZMA stream to repair.
+    """
+    size = fh.seek(0, io.SEEK_END)
+    repaired = OverlayStream(fh, size)
+    fh.seek(0)
+
+    header = fh.read(12)
```

Comment:
```suggestion
    header = fh.read(HEADER_FOOTER_SIZE)
```

or something like this.